app/src/main/java/com/ucm/informatica/spread/Activities/MainTabActivity.java

```diff
@@ -101,5 +143,95 @@ private void setupTabContent(){
         }
     }
 
+    //read coordinates from CoordinatesPolygon.txt
+    private void readPolygonCoordinates() {
+        Point centroid;
+        List<Point> polyCoordList;
+        BufferedReader reader = null;
+        String[] coords;
+        try {
+            reader = new BufferedReader(
+                    new InputStreamReader(getAssets().open("CoordinatesPolygon.txt"), "UTF-8"));
+            String line;
+            while ((line = reader.readLine()) != null) {
+                coords = line.split("\\s+");
+                //first pair is the centroid of the polygon
+                centroid = Point.fromLngLat(Double.parseDouble(coords[1]),Double.parseDouble(coords[0]));
+                polyCoordList = new ArrayList<>();
+                //the rest of pairs are the polygon coordinates
+                for(int i=2; i < coords.length-1; i=i+2) {
+                    polyCoordList.add(Point.fromLngLat(Double.parseDouble(coords[i+1]),Double.parseDouble(coords[i])));
+                }
+                regionMap.put(centroid, new Region(polyCoordList));
+            }
+        } catch (IOException e) {
+            Timber.e(e);
+        } finally {
+            if (reader != null) {
+                try {
+                    reader.close();
+                } catch (IOException e) {
+                    Timber.e(e);
+                }
+            }
+        }
+    }
+
+    public Flashbar getAlertSnackBarGPS(){
```

Comment:
These 3 methods create 3 differents snack bar as described in their library [https://github.com/aritraroy/Flashbar] for confirmation, info and error

toooodo software libre :)